Excalibur dev

ajgdls opened the pull request:
Before they diverge by too much.  This is work in progress so review it at your own risk...

ajgdls committed:
Updating records to use the new dynamic parameter interface.
Added builder command to generate GUI.

ajgdls committed:
Added top level Edm screen for excalibur.

ajgdls committed:
Changed EMPTY_JSON_STRING to be empty quotes.

ajgdls committed:
Add excaliburFemStatusTemplate to builder.

ajgdls committed:
Updated odinDetector to accept array size dynamic parameter names.

ajgdls committed:
Added excalibur screens.

ajgdls committed:
Added excalibur fem status template.

ajgdls committed:
Updated text input box for calibration root to stop it constantly pushing PV values every time the screen was given focus.

ajgdls committed:
Added array counter record.  Updated energy threshold and calibration file root.

ajgdls committed:
Added screen association with the template.

ajgdls committed:
Updated IOC to use different PV names.

ajgdls committed:
Updated detector class to fetch all status.  Added some standard checks for known configuration items.

ajgdls committed:
Added a 2 FEM status template and associated builder code.

ajgdls committed:
Added default status record scan of 1 second.

ajgdls committed:
Added a method for checking the connected status of the Restful API object.

ajgdls committed:
Implemented error message checking.
Updated status loop to read and check for acquisition.

ajgdls committed:
Replaced 2 FEM instances by a 2-FEM instance.

ajgdls committed:
Added record for the number of FEMs present.

ajgdls committed:
Added a calibration status record.

ajgdls committed:
Updated main edm screen to display calibration status.

ajgdls commented:
@GDYendell

GDYendell commented on odinApp/src/odinDetector.cpp:
```diff
@@ -226,18 +239,49 @@ int OdinDetector::createOdinDataParams()
+    // If we are connected to the server then all errors are generated by
+    // the server itself.
+    std::string error_msg = "";
+    mErrorMessage->get(error_msg);
```
The IOC crashes here when I enable status polling. It looks like mErrorMessage is not created.

ajgdls commented:
The only conclusion that I can find is that I've squashed the code when I rebased.  Another reason to split these parts out so re-bases don't keep affecting the same areas of code...
I'll re-implement it now and push the update.

ajgdls committed:
Fixed bug, ensure mErrorMessage parameter is created.

GDYendell committed:
Add mOdinDataParamsCreated check in readStatus

GDYendell commented:
@ajgdls Try it with that new commit. I can now run it with no server and enable status polling and it doesn't crash.

ajgdls commented:
@GDYendell yes that works for me now too.  Thanks.
